<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PatternWindow</class>
 <widget class="QWidget" name="PatternWindow">
  <property name="windowModality">
   <enum>Qt::WindowModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>425</width>
    <height>365</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Record a pattern</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <property name="sizeConstraint">
    <enum>QLayout::SetFixedSize</enum>
   </property>
   <item>
    <layout class="QVBoxLayout" name="mainVerticalLayout">
     <property name="sizeConstraint">
      <enum>QLayout::SetFixedSize</enum>
     </property>
     <item>
      <widget class="QLabel" name="newPatternLabel">
       <property name="font">
        <font>
         <pointsize>18</pointsize>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Record a new Pattern</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="patternNameLayout">
       <item>
        <widget class="QLabel" name="patternNameLabel">
         <property name="text">
          <string>Pattern Name:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="patternNameField"/>
       </item>
      </layout>
     </item>
     <item>
      <widget class="IODevicePlotterWidget" name="ioPlotter"/>
     </item>
     <item>
      <layout class="QHBoxLayout" name="recordingLayout">
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QPushButton" name="toggleRecordingButton">
         <property name="text">
          <string>Start Recording</string>
         </property>
         <property name="checkable">
          <bool>false</bool>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
         <property name="autoDefault">
          <bool>false</bool>
         </property>
         <property name="default">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="playRecordingButton">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Play Recording</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="actionLayout">
       <item>
        <widget class="QPushButton" name="createButton">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Create</string>
         </property>
         <property name="default">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="cancelButton">
         <property name="text">
          <string>Cancel</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>IODevicePlotterWidget</class>
   <extends>QGraphicsView</extends>
   <header location="global">iodeviceplotterwidget.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>patternNameField</tabstop>
  <tabstop>toggleRecordingButton</tabstop>
  <tabstop>playRecordingButton</tabstop>
  <tabstop>createButton</tabstop>
  <tabstop>cancelButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>PatternWindow</sender>
   <signal>recordedChanged(bool)</signal>
   <receiver>playRecordingButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>229</x>
     <y>344</y>
    </hint>
    <hint type="destinationlabel">
     <x>235</x>
     <y>286</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>PatternWindow</sender>
   <signal>playbackChanged(bool)</signal>
   <receiver>toggleRecordingButton</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>102</x>
     <y>341</y>
    </hint>
    <hint type="destinationlabel">
     <x>101</x>
     <y>282</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>PatternWindow</sender>
   <signal>playbackChanged(bool)</signal>
   <receiver>playRecordingButton</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>291</x>
     <y>357</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>293</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>recordingChanged(bool)</signal>
  <signal>recordedChanged(bool)</signal>
  <signal>playbackChanged(bool)</signal>
  <signal>editModeChanged(bool)</signal>
 </slots>
</ui>
